client.go

```diff
@@ -21,74 +21,168 @@ import (
 	"io"
 	"io/ioutil"
 	"net/http"
+	"strings"
 
 	"github.com/mendersoftware/log"
 )
 
 var (
-	errorLoadingClientCertificate = errors.New("Failed to load certificate and key")
-	errorNoServerCertificateFound = errors.New("No server certificate is provided," +
-		" use -trusted-certs with a proper certificate.")
-	errorAddingServerCertificateToPool = errors.New("Adding server certificate " +
-		"to trusted pool failed.")
+	errorLoadingClientCertificate      = errors.New("Failed to load certificate and key")
+	errorNoServerCertificateFound      = errors.New("No server certificate is provided, use -trusted-certs with a proper certificate.")
+	errorAddingServerCertificateToPool = errors.New("Error adding trusted server certificate to pool.")
 )
 
-//TODO: this will be hardcoded for now but should be configurable in future
 const (
-	defaultCertFile         = "/data/certfile.crt"
-	defaultCertKey          = "/data/certkey.key"
-	defaultServerCert       = "/data/server.crt"
-	minimumImageSize  int64 = 4096
+	minimumImageSize int64 = 4096 //kB
 )
 
 type RequestProcessingFunc func(response *http.Response) (interface{}, error)
 
 type Updater interface {
-	GetScheduledUpdate(RequestProcessingFunc, string) (interface{}, error)
+	GetScheduledUpdate(RequestProcessingFunc, string, string) (interface{}, error)
```

Comment:
Makes sense. Also I can see that go library also leaves names of parameters here.